lib/common/metric/MetricFactory.ts

```diff
@@ -435,6 +439,33 @@ export class MetricFactory {
     });
   }
 
+  resolvePeriod(
+    period?: Duration,
+    reportFrequency?: Duration,
+    metricName?: string,
+    errorHandling = ResolvePeriodHandling.ERROR
+  ): Duration {
+    let resolvedPeriod =
```

Comment:
You could declare this as a `const` and just `return reportFrequency` directly when clamping.